cle/backends/java/apk.py

```diff
@@ -73,6 +78,78 @@ def __init__(self, apk_path, binary_stream, entry_point=None, entry_point_params
                                   jni_libs_ld_path=jni_libs_ld_path,
                                   **options)
 
+        # the lifecycle needs to support of pyaxmlparser
+        if PYAXMLPARSER_INSTALLED:
+            self.components = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self.callbacks = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self._set_lifecycle(apk_parser)
```

Comment:
Let's not have classes member conditionally existing. In this case, when `pyaxmlparser` is not installed, `self.components` will not exist. Rather, I think you should set `self.components` (and `self.callbacks`) to `None`.

Reply:
I add the members are ```None``` when else. Your suggestion was helpful.